### DISCUSSION TITLE
Stringify: arrays in one lines for execptions only

### DISCUSSION THREAD

**Discussion by raphaelbastide** [eemeli/yaml#395] [Q&A]
I am wondering if there is any way to output specific keys containing arrays in one line, just like that:
I read the answers for #220 but I don't see how it applies to my case since I am using stringify from Json to YAML. Am I missing something?

What I currently have:
```
config:
  comment: Welcome to été
  currentSynth: tic
  curve:
      a: 0.01
      d: 0.05
      s: 0.001
      r: 0.01
  syncTo: null
  free: false
  recTone: false
patterns:
  - id: 0
    synth: tic
    keys:
    - id: 1
      note: 850
      offset: 81.5
      duration: 530
```

What I am trying to do:

```
config:
  comment: Welcome to été
  currentSynth: tic
  curve: [a: 0.01, d: 0.05, s: 0.001, r: 0.01]
  syncTo: null
  free: false
  recTone: false
patterns:
  - id: 0
    synth: tic
    keys:
    - id: 1
      note: 850
      offset: 81.5
      duration: 530
```

**Comment by eemeli** [comment]
Collection nodes have a boolean `flow` property that you can set to `true`. With that set, if the contents are short enough, they'll be output on a single line. See here for instructions: https://eemeli.org/yaml/#modifying-nodes

Also, I'm presuming here that you're looking to keep the collection as a map, which would have this representation:
```yaml
{ a: 0.01, d: 0.05, s: 0.001, r: 0.01 }
```

**Reply by raphaelbastide** [reply]
Thank you, so if I understand correctly, in order to add a `flow:true` to some of my values, I need to split `YAML.stringify()` into `YAML.parseDocument()` + `YAML.visit()` +  `string()`, is that right?

**Reply by eemeli** [reply]
In principle, yes, but with two caveats:
1. If your input data is already a JS object, you may want to use `new YAML.Document(value)` to create the doc.
2. Depending on how you identify the node to use flow-collection syntax for, it may be easier to use `doc.getIn()` to find it rather than `YAML.visit()`.

**Reply by raphaelbastide** [reply]
I have difficulties to use `getin()`. I tried to test the same code as the doc’s example, but it returns undefined:

```
 const doc = new YAML.Document({ a: 1, b: [ 2, 5 ] })
 console.log(doc.getIn(['b', 1])); // undefined
```

**Reply by eemeli** [reply]
`console.log()` always returns undefined. With your example, it does successfully print out `5` for me before returning `undefined`.

**Reply by raphaelbastide** [reply]
That’s strange, maybe I missed something, could you take a look at this ? https://jsfiddle.net/20qwhoab/1/

**Reply by eemeli** [reply]
Looks like the `yaml-playground` hasn't been updated for a while, and the file you're pulling in is a `yaml@1` build. It is not meant to be used externally like that. If you use a `yaml@2` build, you'll get the expected results.

**Reply by raphaelbastide** [reply]
I see, thanks for pointing this out. I am looking for a way to have an old school single JavaScript file containing the YAML library, to include to my project: for some reasons*, I prefer to avoid the npm environment. Could you point me a way to generate such a file? I struggle to find it on CDNs or to bundle it with cli tools.
_*_ this project has to work in the browser without installation

**Reply by raphaelbastide** [reply]
Hi, I understand my request can be a bit strange, but I am still stuck trying to find a single-file version of `yaml@2`. Do you think it is something you can help me with?

**Reply by eemeli** [reply]
You should use a bundler of your choice to transpile & polyfill the npm package for the set of browsers that you need to support and the module format that you wish to use. I tend to use [Rollup](https://rollupjs.org/guide/en/).

**Reply by raphaelbastide** [reply]
Thank you for your answer, I found the solution using Rollup!
